app/Http/Controllers/Bible/BibleFilesetsDownloadController.php

```diff
@@ -101,20 +101,27 @@ function () use ($fileset_id, $type) {
             );
         }
 
-        $cache_params = $this->removeSpaceFromCacheParameters([$fileset_id, $book_id, $chapter, $key, $limit, $type]);
+        $bulk_access_control = $this->allowedForDownload($fileset);
+
+        if (isset($bulk_access_control->original['error'])) {
+            return $bulk_access_control;
+        }
+
+        $cache_params = $this->removeSpaceFromCacheParameters([
+            $fileset_id,
+            $book_id,
+            $chapter,
+            $key,
+            $limit,
+            $type,
+            $hasUserLoggedIn
```

Comment:
@bradflood, I added the $hasUserLoggedIn value to the cache parameters because the access group list can change depending on whether the user is logged in or he is not for bible.is.